lib/sailthru.js

```diff
@@ -179,6 +189,16 @@
           var protocol = _url.port && _url.port === 80 ? 'http:' : 'https:';
           this.api_url = protocol + '//' + this.api_url;
       }
+      // Validate proxy
+      if (this.proxy_url) {
+        var _proxy = url.parse(this.proxy_url);
+        var proxy_protocol_supported = ["http:", "https:", "socks:", "socks4:", "socks5:", "pac+http:"].includes(_proxy.protocol);
+        var proxy_used_without_protocol = _proxy.protocol == null;
+        if (_proxy.protocol && !proxy_protocol_supported || proxy_used_without_protocol) {
```

Comment:
doesn't line 195 cover for `proxy_used_without_protocol`? that should return false if `proxy.protocol` is null because it's not in the array.

Reply:
Yeah, you're right. This was an oversight when I was refactoring an initial solution. Nice catch.